### PR TITLE
Zbs247/integrate k fit into sigproc v2

### PR DESCRIPTION
**Pull request by zack-erisyon**
Monsterous PR
Implements nn_v2 changes to use row fitting and converts nn-v2 to a new ctype way of doing things that sim_v2 and survey will also move to.

### CODE REVIEW

File: plaster/run/nn_v2/c/nn_v2.py [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+    use_row_k_p_val=True,
+):
+    """
+    with nn_v2.context(...) as ctx:
+        zap.work_orders(do_classify_radrows, ...)
+    """
+    lib = load_lib()
+
+    check.t(gain_model, GainModel)
+    assert gain_model.n_channels == 1  # Temporary
```
Comment (danwinkler) [inline plaster/run/nn_v2/c/nn_v2.py:198]: Is this still supposed to be here?

**Review by danwinkler — APPROVED** [review]
In general the ctypes stuff looks fine, I think after we build a few more modules with ctypes we'll have a good feel about how to DRY it out, etc.

Definitely an improvement over cython IMO.

File: plaster/run/nn_v2/c/nn_v2.py (hunk 1 above)
Comment (zack-erisyon) [reply plaster/run/nn_v2/c/nn_v2.py:198]: Thanks for calling that out, but yes, it is there until future when I add more channels.

File: plaster/run/nn_v2/c/build.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
+from plumbum import local, FG
```
Comment (ossareh) [inline plaster/run/nn_v2/c/build.py:1]: Thanks for stepping in @danwinkler !

@zack-erisyon I don't really know enough about what I'm about to ask for; but this seems the right time to spend a bit of effort on this experiment: Can we statically link flann at this point? If so, we can get rid of the need to copy it in our docker builds, and remove the setting of LD_LIBRARY_PATH.

I looked into it and gcc seems to mostly provide switches for statically including libstdc++, etc. 

I found this on the subject: https://medium.com/@meghamohan/all-about-static-libraries-in-c-cea57990c495

It shows the use of two tools new to me:

 * `ar`
 * `ranlib*

`ar` produces an archive of libraries, and `ranlib` seems optional but it depends on what you use to create the archive; in any case, it updates the index. With the archive created it seems we can then just `gcc -L <build dir> -l<name of archive>` and we get the archive statically compiled into the resulting library.

I'm happy to take a stab at making this work, but I'd need some time of yours to orient me in the project, build flows, etc.

Comment (zack-erisyon) [reply plaster/run/nn_v2/c/build.py:1]: I think that's a good idea. I've added a task in my column on it. I don't want to tackle it at moment.

**Pull request merged**